modules/apps/adaptive-media/adaptive-media-content-transformer-impl/src/main/java/com/liferay/adaptive/media/content/transformer/internal/HtmlContentTransformerImpl.java

```diff
@@ -39,42 +37,90 @@ public String transform(String html) throws PortalException {
 			return null;
 		}
 
-		if (!html.contains(AMImageHTMLConstants.ATTRIBUTE_NAME_FILE_ENTRY_ID)) {
-			return html;
-		}
+		StringBundler sb = new StringBundler();
 
-		return super.transform(html);
-	}
+		int i = 0;
 
-	@Override
-	protected FileEntry getFileEntry(Matcher matcher) throws PortalException {
-		long fileEntryId = GetterUtil.getLong(matcher.group(1));
+		while (i < html.length()) {
+			int imgStart = html.indexOf(_OPEN_TAG_TOKEN_IMG, i);
 
-		return _dlAppLocalService.getFileEntry(fileEntryId);
-	}
+			if (imgStart == -1) {
+				sb.append(html.substring(i));
 
-	@Override
-	protected Pattern getPattern() {
-		return _pattern;
-	}
+				break;
+			}
 
-	@Override
-	protected String getReplacement(String originalImgTag, FileEntry fileEntry)
-		throws PortalException {
+			int imgEnd = html.indexOf(
```

Comment:
Suggestion

imgEnd & imaStart  could be renamed to imgTagEnd and imgTagStart

Reply:
I used `imgStart` here as well: https://github.com/brianchandotcom/liferay-portal/blob/8af1d7bc80cd376113e15ce737760d5db03f2b66/modules/apps/adaptive-media/adaptive-media-image-content-transformer-backwards-compatibility/src/main/java/com/liferay/adaptive/media/image/content/transformer/backwards/compatibility/internal/AMBackwardsCompatibilityHtmlContentTransformer.java

Let's wait for the other review and I'll change it if necessary.